cmd/kube-proxy/app/server.go

```diff
@@ -399,6 +399,20 @@ func NewProxyServer(config *componentconfig.KubeProxyConfiguration, cleanupAndEx
 	proxyMode := getProxyMode(string(config.Mode), iptInterface, iptables.LinuxKernelCompatTester{})
 	if proxyMode == proxyModeIPTables {
 		glog.V(0).Info("Using iptables Proxier.")
+		var nodeIP net.IP
+		if config.BindAddress == "0.0.0.0" || config.BindAddress == "" {
+			nodeIP = getNodeIP(client, hostname)
+		} else {
+			nodeIP = net.ParseIP(config.BindAddress)
+			if nodeIP == nil {
+				return nil, fmt.Errorf("bind-address %s must be valid ip", config.BindAddress)
+			}
+			if local, err := isLocalIP(nodeIP.String()); err != nil {
```

Comment:
Thanks Tim,  ptables.proxier would only check `nodeip==nil`,  see: https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/iptables/proxier.go#L471,  I think we need check bindaddress if valide,  could you tell me what is a good-enouth error.